model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/tool/ExperimentalMessagesUtils.java

```diff
@@ -0,0 +1,144 @@
+package io.quarkiverse.langchain4j.ollama.tool;
+
+import static dev.langchain4j.data.message.ContentType.IMAGE;
+import static dev.langchain4j.data.message.ContentType.TEXT;
+import static java.util.Collections.singletonList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.data.message.*;
+import io.quarkiverse.langchain4j.data.AiStatsMessage;
+import io.quarkiverse.langchain4j.ollama.ImageUtils;
+import io.quarkiverse.langchain4j.ollama.Message;
+import io.quarkiverse.langchain4j.ollama.Role;
+
+class ExperimentalMessagesUtils {
```

Comment:
Javadoc please :)

And perhaps a link to whatever describes these formats

Reply:
done